### ISSUE TITLE
`jest/require-hook` triggers on non test file

### ISSUE THREAD

**Issue by thibaultdelor** [jest-community/eslint-plugin-jest#952]
Upgrading from `24.5.0` to `25.2.0`, I now have:

```
<project_folder>/src/index.ts
  67:1  error  This should be done within a hook  jest/require-hook
```

`index.ts` doesn't contain any tests, doesn't import jest and the method on which it complains is the one who initalise my server:
```typescript
async function initApolloServer() {
  // init some stuff...
}

initApolloServer().then((apolloRouter) => {    // <----- This is the offending line
  // start the server...
});
```
